pkg/ruler/base/client_pool.go

```diff
@@ -48,27 +48,29 @@ func newRulerClientPool(clientCfg grpcclient.Config, logger log.Logger, reg prom
 	})
 
 	return &rulerClientsPool{
-		client.NewPool("ruler", poolCfg, nil, newRulerClientFactory(clientCfg, reg), clientsCount, logger),
+		client.NewPool("ruler", poolCfg, nil, newRulerClientFactory(clientCfg, reg, metricsNamespace), clientsCount, logger),
```

Comment:
It feels weird to me that we're passing in the metrics namespace all over the codebase. Feels like we should be able to solve this with `prometheus.WrapRegistererWithPrefix("loki_", reg)` or defining the namespace in one central location as a singleton.

Reply:
I remember @MichelHollands tried to make `WrapRegistererWithPrefix` work. I'll check with him and do this change in a follow-up

Reply:
Cool, works for me